plugins/embed-optimizer/class-embed-optimizer-tag-visitor.php

```diff
@@ -26,26 +26,76 @@ final class Embed_Optimizer_Tag_Visitor {
 	 */
 	protected $added_lazy_script = false;
 
+	/**
+	 * Determines whether the processor is currently at a figure.wp-block-embed tag.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_HTML_Tag_Processor $processor Processor.
+	 * @return bool Whether at the tag.
+	 */
+	private function is_embed_figure( OD_HTML_Tag_Processor $processor ): bool {
+		return (
+			'FIGURE' === $processor->get_tag()
+			&&
+			true === $processor->has_class( 'wp-block-embed' )
+		);
+	}
+
+	/**
+	 * Determines whether the processor is currently at a div.wp-block-embed__wrapper tag.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_HTML_Tag_Processor $processor Processor.
+	 * @return bool Whether the tag should be measured and stored in URL metrics
+	 */
+	private function is_embed_wrapper( OD_HTML_Tag_Processor $processor ): bool {
+		return (
+			'DIV' === $processor->get_tag()
+			&&
+			true === $processor->has_class( 'wp-block-embed__wrapper' )
+		);
+	}
+
 	/**
 	 * Visits a tag.
 	 *
 	 * @since 0.2.0
 	 *
 	 * @param OD_Tag_Visitor_Context $context Tag visitor context.
-	 * @return bool Whether the visit or visited the tag.
+	 * @return bool Whether the tag should be measured and stored in URL metrics.
 	 */
 	public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 		$processor = $context->processor;
-		if ( ! (
-			'FIGURE' === $processor->get_tag()
-			&&
-			true === $processor->has_class( 'wp-block-embed' )
-		) ) {
+
+		/*
+		 * The only thing we need to do if it is a div.wp-block-embed__wrapper tag is return true so that the tag
+		 * will get measured and stored in the URL Metrics.
+		 */
+		if ( $this->is_embed_wrapper( $processor ) ) {
+			return true;
+		}
+
+		// Short-circuit if not a figure.wp-block-embed tag.
+		if ( ! $this->is_embed_figure( $processor ) ) {
 			return false;
 		}
 
-		$max_intersection_ratio = $context->url_metrics_group_collection->get_element_max_intersection_ratio( $processor->get_xpath() );
+		$embed_wrapper_xpath = $processor->get_xpath() . '/*[1][self::DIV]';
+		$minimum_height      = $context->url_metrics_group_collection->get_element_minimum_height( $embed_wrapper_xpath );
+		if ( is_float( $minimum_height ) ) {
+			$style = $processor->get_attribute( 'style' );
+			if ( is_string( $style ) ) {
+				$style = rtrim( trim( $style ), ';' ) . '; ';
+			} else {
+				$style = '';
+			}
+			$style .= sprintf( 'min-height: %dpx;', $minimum_height );
```

Comment:
Given that heights are floats, should this rather be:

```suggestion
			$style .= sprintf( 'min-height: %fpx;', $minimum_height );
```

But then it could result in some very long decimals, like `min-height: 500.200000px;`